genesyscloud/resource_genesyscloud_processautomation_trigger.go

```diff
@@ -0,0 +1,583 @@
+package genesyscloud
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/mypurecloud/platform-client-sdk-go/v56/platformclientv2"
+)
+
+type ProcessAutomationTrigger struct {
+	Id              *string             `json:"id,omitempty"`
+	TopicName       *string             `json:"topicName,omitempty"`
+	Name            *string             `json:"name,omitempty"`
+	Target          *Target             `json:"target,omitempty"`
+	MatchCriteria   *[]MatchCriteria    `json:"matchCriteria,omitempty"`
+	Enabled         *bool               `json:"enabled,omitempty"`
+	EventTTLSeconds *int                `json:"eventTTLSeconds,omitempty"`
+	Version         *int                `json:"version,omitempty"`
+}
+
+type UpdateTriggerInput struct {
+	Name            *string             `json:"name,omitempty"`
+	Target          *Target             `json:"target,omitempty"`
+	MatchCriteria   *[]MatchCriteria    `json:"matchCriteria,omitempty"`
+	Enabled         *bool               `json:"enabled,omitempty"`
+	EventTTLSeconds *int                `json:"eventTTLSeconds,omitempty"`
+	Version         *int                `json:"version,omitempty"`
+}
+
+type MatchCriteria struct {
+    JsonPath        *string             `json:"jsonPath,omitempty"`
+    Operator        *string             `json:"operator,omitempty"`
+    Value           *string             `json:"value,omitempty"`
+    Values          *[]string           `json:"values,omitempty"`
+}
+
+type Target struct {
+    Type        *string             `json:"type,omitempty"`
+    Id          *string             `json:"id,omitempty"`
+}
+
+var (
+	target = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Description: "Type of the target the trigger is configured to hit",
+				Type:           schema.TypeString,
+				Required:       true,
+			},
+			"id": {
+				Description:  "Id of the target the trigger is configured to hit",
+				Type:           schema.TypeString,
+				Required:       true,
+			},
+		},
+	}
+
+	matchCriteria = &schema.Resource{
+        Schema: map[string]*schema.Schema{
+            "json_path": {
+                Description: "The json path of the topic event to be compared to match criteria value",
+                Type:           schema.TypeString,
+                Required:       true,
+            },
+            "operator": {
+                Description:  "The operator used to compare the json path against the value of the match criteria",
+                Type:           schema.TypeString,
+                Required:       true,
```

Comment:
You can add a validationFunc for `validations.StringInSlice` and check for `"GreaterThanOrEqual",
                        "LessThanOrEqual",
                        "Equal",
                        "NotEqual",
                        "LessThan",
                        "GreaterThan",
                        "NotIn",
                        "In",
                        "Contains",
                        "All",
                        "Exists",
                        "Size"`